### ISSUE TITLE
some settings need to be exposed to the user for ideal functionality

### ISSUE THREAD

**Issue by whiteowl3** [jellyfin-archive/jellyfin-roku-legacy#550]
**Describe the feature you'd like**
a settings menu needs to be implemented somewhere that allows for certain program-wide settings to be set.
at issue for me at the moment:

- drop graphical subtitles or transcode them
- output hdr or transcode it
- passthrough dd+ or transcode it
- initialize client certificates or dont
- pass dev-id header or dont
- URL to use if we receive local http links (for now assume single port rp)

but there will probably be others.
all input on where that menu should be or anything else it should include is welcome and appreciated.

**Comment by debmint** [comment]
Possibly related to this issue:
I was just thinking this morning about how nice it would be for a user to set preferences which the server allows the user to set, such as order in which libraries are shown, etc.

**Comment by neilsb** [comment]
I have the beginnings of a user preferences system done. Defined by a json file to make it easy to add etc. 

Needs a little UI work and there is a focus issue, but other that it's functional.

Only set up to work with Boolean values (as that's what I was needing it for) but easy to add other types. Was going to add number, text and select as they were needed. 

I'll try get it finished off and a PR submitted.

**Comment by whiteowl3** [comment]
- ~~drop graphical subtitles or transcode them~~ #675 
- ~~output hdr or transcode it~~ #718 
- passthrough dd+ or transcode it
- ~~initialize client certificates or dont~~
- ~~pass dev-id header or dont~~ #677 
- URL to use if we receive local http links (for now assume single port rp)
- URL to download a CA from (warn user if not https)

**Comment by cewert** [comment]
So the way this issue is written it will never get closed because you have so much going on in here. Can we break these up in separate issues? It would give you a chance to explain what they all do and why you think they're each important. And it would be nice to have a dedicated place to talk about each of these. Best way to implement them, relevant code etc. etc.

**Comment by whiteowl3** [comment]
in my mind this issue is just waiting on the settings menu to be implemented, and could just be closed and replaced with more specific issues afterwards, but I don't see the problem that you are describing.  certainly it could be broken out into a discussion or series of discussions ("core", "MTLS", and "strm" could be useful subdivisions) but, really i see no reason why we can't just discuss it here and then close the issue when everything is either implemented or rejected.

**Comment by whiteowl3** [comment]
oh it looks like neil dropped his user settings menu in #583, quite a few weeks ago.

**Comment by whiteowl3** [comment]
At this point I don't even remember why I wanted to force dd+ passthrough, so I will probably nix that if I can't recall.  Other than that, everything else is just waiting on #668
